core/block/editor/state/position.go

```diff
@@ -185,6 +185,16 @@ func (s *State) setChildrenIds(parent *model.Block, childrenIds []string) {
 	s.addCacheIds(parent, childrenIds...)
 }
 
+// do not use this method outside of normalization
+func (s *State) SetChildrenIds(parent *model.Block, childrenIds []string) {
```

Comment:
Maybe we can leave only this special public method?
We can perform check on children list change and remove required ids from cache.
RemoveFromCache is unnecessary then